src/widgetastic/browser.py

```diff
@@ -336,17 +337,15 @@ def element(self, locator, *args, **kwargs):
         """
         try:
             vcheck = self._locator_force_visibility_check(locator)
-            if vcheck is not None:
-                kwargs['check_visibility'] = vcheck
+            self.logger.error(vcheck)
+            kwargs['check_visibility'] = vcheck or kwargs.get('check_visibility', False)
```

Comment:
I think I know why the tests fail. If `vcheck` is `False` and `kwargs.get('check_visibility')` is ` True` than `True` will be assigned to  `kwargs['check_visibility']`. But `vcheck` should be always prefered according to the test https://github.com/RedHatQE/widgetastic.core/blob/master/testing/test_browser.py#L233. So in order not to execute `is_displayed` twice is enough just remove `if` block:
```python
try:
    vcheck = self._locator_force_visibility_check(locator)
    if vcheck is not None:
        kwargs['check_visibility'] = vcheck
    elements = self.elements(locator, *args, **kwargs)
    return elements[0]
except IndexError:
    raise NoSuchElementException(
        'Could not find an element {}'.format(repr(locator))) from None
```
^ This patch breaks `test_element_visible_after_invisible_and_classes_and_execute_script` and I'm sure that test is not correct. It should expect another value:

```python
def test_element_visible_after_invisible_and_classes_and_execute_script(browser):
    assert 'invisible' in browser.classes('//div[@id="visible_invisible"]/p', check_visibility=False)
```